libs/api-client-tiger/src/metadataUtilities.ts

```diff
@@ -0,0 +1,156 @@
+    JsonApiFactList,
+    JsonApiFilterContextList,
+    jsonApiHeaders,
+    JsonApiLabelList,
+    JsonApiMetricList,
+    JsonApiOrganizationList,
+    JsonApiVisualizationObjectList,
+    JsonApiWorkspaceList,
+} from "./index";
+
+const DefaultPageSize = 250;
+const DefaultOptions = {
+    headers: jsonApiHeaders,
+    query: {
+        size: DefaultPageSize,
+    },
+};
+
+function createOptionsForPage(page: number, options: MetadataGetEntitiesOptions): MetadataGetEntitiesOptions {
+    return merge({}, DefaultOptions, options, { query: { page } });
+}
+
+/**
+ * Common parameters for all API client getEntities* parameters.
+ *
+ * Note: the different generated client functions are actually incorrect. They list page, size, include, sort in
+ * the params but they are not picked from there anyway. They need to be passed in options as query parameters.
+ *
+ * @internal
+ */
+export type MetadataGetEntitiesParams = {
+    workspaceId: string;
+};
+
+/**
+ * Common parameters for all API client getEntities* parameters.
+ *
+ * @internal
+ */
+export type MetadataGetEntitiesOptions = {
+    headers?: object;
+    query?: {
+        page?: number;
+        size?: number;
+        include?: any;
+        sort?: any;
+    };
+};
+
+/**
+ * All possible responses of API client getEntities* functions.
+ *
+ * @internal
+ */
+export type MetadataGetEntitiesResult =
+    | JsonApiOrganizationList
+    | JsonApiWorkspaceList
+    | JsonApiVisualizationObjectList
+    | JsonApiAnalyticalDashboardList
+    | JsonApiDatasetList
+    | JsonApiAttributeList
+    | JsonApiLabelList
+    | JsonApiMetricList
+    | JsonApiFactList
+    | JsonApiFilterContextList;
+
+/**
+ * All API client getEntities* functions follow this signature.
+ *
+ * @internal
+ */
+export type MetadataGetEntitiesFn<T extends MetadataGetEntitiesResult> = (
+    params: MetadataGetEntitiesParams,
+    options: MetadataGetEntitiesOptions,
+) => AxiosPromise<T>;
+
+/**
+ * Tiger metadata utility functions
+ *
+ * @internal
+ */
+export class MetadataUtilities {
+    /**
+     * Given a function to get a paged list of metadata entities, API call parameters and options, this function will
+     * retrieve all pages from the metadata.
+     *
+     * The parameters are passed to the function as is. The options will be used as a 'template'. If the options specify
+     * page `size`, it will be retained and used for paging. Otherwise the size will be set to a default value (250). The
+     * `page` number will be added dynamically upon each page request.
+     *
+     * @param client - API client to use, this is required so that function can correctly bind 'this' for
+     *  the entitiesGet function
+     * @param entitiesGet - function to get pages list of entities
+     * @param params - parameters accepted by the function
+     * @param options - options accepted by the function
+     * @internal
+     */
+    public static getAllPagesOf = async <T extends MetadataGetEntitiesResult>(
```

Comment:
What about type it like this (to preserve inference in each client)?
```
public static getAllPagesOf = async <
        T extends MetadataGetEntitiesResult,
        P extends MetadataGetEntitiesParams
    >(
        client: ITigerClient,
        entitiesGet: MetadataGetEntitiesFn<T, P>,
        params: P,
        options: MetadataGetEntitiesOptions = {},
    ): Promise<T[]> 
```

Reply:
done